### PR TITLE
Update MeshTools

### PR DESCRIPTION
**Pull request by tridelat**
PR for issue #849

### CODE REVIEW

**Comment by tridelat** [comment]
@cekees I added a partitioning.pxd file so I can import functions from partitioning.h of #860. The current problem I have is that partitioning.h does not compile as some dependencies are missing.

**Comment by cekees** [comment]
@tridelat I just pushed some updates that finish refactoring the partitioning into partitioning.h and partitioning.cpp.  I guess our tasks are sort of coupled, since I need your mesh.pxd to actually do the wrappers of the partitioning functions, which need to do conversion to/from the `CMesh` extension type and the C struct `Mesh`.

File: proteus/cmeshToolsTest.pyx [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+from libcpp cimport bool
+from mpi4py import MPI
+cimport mesh as cppm
+cimport flcbdfWrappersModule as flcbdWrappersModule
+from proteus import Comm
+
+cdef class CMeshLink:
+    """
+    Need to have this intermediate class for CMesh in order to make CMesh a
+    pickable class..
```
Comment (cekees) [inline proteus/cmeshToolsTest.pyx]: Do you mean pickleable here? Do we need to be able to pickle CMesh objects in order to ensure that Mesh objects themselves can be pickled? Just interested in what you're doing here.

**Comment by cekees** [comment]
closing this PR to squash and merge onto a new branch

**Pull request closed without merging**